runtime/node/src/shared/helpers.ts

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import fs from 'fs';
+import minimist from 'minimist';
+import path from 'path';
+import { Server } from 'restify';
+import merge from 'lodash/merge';
+import {
+  BotFrameworkAdapter,
+  BotFrameworkAdapterSettings,
+  ChannelServiceRoutes,
+  ConversationState,
+  InputHints,
+  SkillHandler,
+  TurnContext,
+  UserState,
+  WebRequest,
+  WebResponse,
+} from 'botbuilder';
+import { AuthenticationConfiguration, SimpleCredentialProvider } from 'botframework-connector';
+import { ComposerBot } from './composerBot';
+import { BotSettings } from './settings';
+import { SkillConversationIdFactory } from './skillConversationIdFactory';
+
+/**
+ * Get listening port for listening from environment variables or arguments.
+ */
+export const getServerPort = () => {
+  const argv = minimist(process.argv.slice(2));
+  // prefer the argv port --port=XXXX over process.env because the parent Composer app uses that.
+  const port = argv.port || process.env.port || process.env.PORT || 3979;
+  return port;
+};
+
+/**
+ * Get path of bot project.
+ */
+export const getProjectRoot = (): string => {
+  // get the root folder according to environment
+  if (process.env.node_environment === 'production') {
+    return path.join(__dirname, '../../azurewebapp/ComposerDialogs');
+  } else {
+    return path.join(__dirname, '../../../');
+  }
+};
+
+/**
+ * Get bot settings from configuration file, generated luis configuration or arguments.
+ * @param projectRoot Root path of bot project.
+ */
+export const getSettings = (projectRoot = getProjectRoot()): BotSettings => {
+  // Find settings json file
+  let settings = {} as BotSettings;
+  // load appsettings.json
+  const appsettingsPath = path.join(projectRoot, 'settings/appsettings.json');
+  if (fs.existsSync(appsettingsPath)) {
+    settings = JSON.parse(fs.readFileSync(appsettingsPath, 'utf8'));
+  }
+
+  // load generated settings
+  const generatedPath = path.join(projectRoot, 'generated');
+  if (fs.existsSync(generatedPath)) {
+    const generatedFiles = fs.readdirSync(generatedPath);
+    for (const file of generatedFiles) {
+      if (file.endsWith('.json')) {
+        const items = JSON.parse(fs.readFileSync(path.join(generatedPath, file), 'utf8'));
+        settings.luis = merge(settings.luis, items.luis); // merge luis settings
```

Comment:
```suggestion
        settings.luis = merge({}, settings.luis, items.luis); // merge luis settings
```

`merge` mutates the source object.